File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/PersonController.vb
Public Class PersonController
    Inherits System.Web.Mvc.Controller


    Function Index() As ActionResult
        Return View("Index")
    End Function

    Function Details(ByVal id As Integer?) As ActionResult
        If Not id.HasValue Then
            Return RedirectToAction("Index")
        End If

        Return View("Details")
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/PersonController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class PersonController : Controller
    {
        public ActionResult Index()
        {
            return View("Index");
        }

        public ActionResult Details(int? id)
        {
            if (!id.HasValue)
                return RedirectToAction("Index");
            return View("Details");
        }

    }
}
